src/sbl_filing_api/routers/filing.py

```diff
@@ -317,6 +324,13 @@ async def put_contact_info(request: Request, lei: str, period_code: str, contact
 )
 @requires("authenticated")
 async def get_latest_submission_report(request: Request, lei: str, period_code: str):
+    filing = await repo.get_filing(request.state.db_session, lei, period_code)
+    if not filing:
+        raise RegTechHttpException(
+            status_code=status.HTTP_404_NOT_FOUND,
+            name="Filing Not Found",
+            detail=f"There is no Filing for LEI {lei} in period {period_code}, unable to get latest submission for it.",
+        )
```

Comment:
no filing scenario for this endpoint isn't covered, let's add test for this as well

Reply:
I think I did add test for no filing scenario.
`async def test_get_latest_sub_report(
        self, mocker: MockerFixture, app_fixture: FastAPI, get_filing_mock: Mock, authed_user_mock: Mock
    ):
        sub_mock = mocker.patch("sbl_filing_api.entities.repos.submission_repo.get_latest_submission")
        sub_mock.return_value = SubmissionDAO(
            id=1,
            submitter=UserActionDAO(
                id=1,
                user_id="1234-5678-ABCD-EFGH",
                user_name="Test Submitter User",
                user_email="test1@cfpb.gov",
                action_type=UserActionType.SUBMIT,
                timestamp=datetime.datetime.now(),
            ),
            filing=1,
            state=SubmissionState.VALIDATION_IN_PROGRESS,
            validation_ruleset_version="v1",
            submission_time=datetime.datetime.now(),
            filename="file1.csv",
        )

        file_content = "Test"
        file_mock = mocker.patch("sbl_filing_api.services.submission_processor.get_from_storage")
        file_mock.return_value = [c for c in file_content]

        client = TestClient(app_fixture)
        res = client.get("/v1/filing/institutions/1234567890ZXWVUTSR00/filings/2024/submissions/latest/report")
        sub_mock.assert_called_with(ANY, "1234567890ZXWVUTSR00", "2024")
        file_mock.assert_called_with("2024", "1234567890ZXWVUTSR00", "1" + submission_processor.REPORT_QUALIFIER)
        assert res.status_code == 200
        assert res.text == "Test"
        assert res.headers["content-type"] == "text/csv; charset=utf-8"
        assert res.headers["content-disposition"] == 'attachment; filename="1_validation_report.csv"'

        sub_mock.return_value = []
        client = TestClient(app_fixture)
        res = client.get("/v1/filing/institutions/1234567890ZXWVUTSR00/filings/2024/submissions/latest/report")
        sub_mock.assert_called_with(ANY, "1234567890ZXWVUTSR00", "2024")
        assert res.status_code == 204

        **# verify Filing Not Found RegTechHttpException returned when filing does not exist
        get_filing_mock.return_value = None
        client = TestClient(app_fixture)
        res = client.get("/v1/filing/institutions/1234567890ZXWVUTSR00/filings/2024/submissions/latest")
        assert res.status_code == 404**
`

Reply:
ah~, I see you added in there, except the endpoint is still only `/latest`, you didn't have `/report` after the `/latest`

Reply:
Oops, I see. Fixing it.